File: yuv_convert/src/main/java/com/zmy/yuv_convert/output/PackedByteArrayOutput.java
package com.zmy.yuv_convert.output;

import com.zmy.yuv_convert.Format;

public class PackedByteArrayOutput extends Output<byte[]> {
    byte array[];

    public PackedByteArrayOutput(Format format) {
        super(format);
    }

    @Override
    public byte[] getOutput() {
        if (array == null || array.length < format.getMinFrameSize(width, height)) {
            array = new byte[format.getMinFrameSize(width, height)];
        }
        container.position(0);
        container.limit(container.capacity());
        container.get(array);
        return array;
    }
}

File: yuv_convert/src/main/java/com/zmy/yuv_convert/output/PackedByteArrayOutput.kt
package com.zmy.yuv_convert.output

import com.zmy.yuv_convert.Format

class PackedByteArrayOutput(format: Format) : Output<ByteArray>(format) {
    var array: ByteArray? = null
    override fun getOutput(): ByteArray {
        if ((array?.size ?: -1) < format.getMinFrameSize(width, height)) {
            array = ByteArray(format.getMinFrameSize(width, height))
        }
        container?.position(0)
        container?.let { it.limit(it.capacity()) }
        container?.get(array!!)
        return array!!
    }
}